### PR TITLE
Address test warnings by pinning `pyqt5-sip<12.13.0` and updating `setup.cfg` ignore entries (`pkg_resources`)

### LINKED ISSUE #4354: Address new test warnings (and double-check the existing ignore list)
Our test suite is reporting new warnings, so I wanted to add a maintenance TODO for myself.

### PR DESCRIPTION
**Pull request by joshuacwnewton**
## Description

This is a small PR to revisit the warnings caught by our test suite:

- pyqt5-sip: See https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/4354
- pkg_resources deprecation --> ~~fixed upstream~~ Not true! `wandb` still uses `pkg_resources`

## Linked issues

Fixes #4354.

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -43,11 +43,6 @@ filterwarnings =
     ignore:.*in an upcoming release, it will be required to pass the indexing argument.*:UserWarning:torch.*:
```
Comment (joshuacwnewton) [inline setup.cfg:43]: I think we could do a little more work with the `torch` related warnings, but I think we might be able to tackle those as part of https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4345, given that we're upgrading torch in that PR.

**Comment by joshuacwnewton** [comment]
Removing this from the 6.2 milestone since it is not critical to review this prior to 6.2.

**Review by mguaypaq — APPROVED** [review]
Looks good, thanks for this bit of maintenance!

File: requirements.txt
[the remark this answers is not in the document]
Comment (joshuacwnewton) [reply requirements.txt:1]: Addressed in [82b23fc](https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4355/commits/82b23fc66ed7f216f9372c20821b8a51a17e7b1b).

File: setup.cfg (hunk 1 above)
Comment (joshuacwnewton) [reply setup.cfg:44]: Updated in https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4355/commits/e8c2cee755f995d1aaf3781329357dfae7610dea.

**Pull request merged**